### PR TITLE
WIP WS Correction

### PR DESCRIPTION
**Pull request by as2388**
WIP WS Correction. The core of the work is done, just need to:
 - Assign CODING_ERROR codes where the normal/WS codes conflict, rather than trusting the WS code.
- Handle multi-project datasets (i.e. CSAP Somalia).

Requires https://github.com/AfricasVoices/CoreDataModules/pull/139

### CODE REVIEW

File: generate_outputs.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import argparse
 import json
 import os
+import random
```
Comment (lukechurch) [inline generate_outputs.py]: Is this used? I don't see it in the diff

File: src/lib/pipeline_configuration.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ class CodeSchemes(object):
 
 class CodingPlan(object):
     def __init__(self, raw_field, coded_field, coda_filename, cleaner=None, code_scheme=None, time_field=None,
-                 run_id_field=None, icr_filename=None, analysis_file_key=None, id_field=None,
+                 run_id_field=None, icr_filename=None, analysis_file_key=None, id_field=None, ws_code=None,
```
Comment (lukechurch) [inline src/lib/pipeline_configuration.py:48]: ws_code or ws_codeID? If I understand correctly these are the IDs associated with the label in the coda scheme that indicates this target? (Just that sentence itself is slightly absurd)

File: src/lib/pipeline_configuration.py [hunk 3]
```diff
@@ -74,6 +75,7 @@ class PipelineConfiguration(object):
                    icr_filename="s02e01.csv",
                    run_id_field="rqa_s02e01_run_id",
                    analysis_file_key="rqa_s02e01_",
+                   ws_code=CodeSchemes.WS_CORRECT_DATASET.get_code_with_match_value("s02e01"),
```
Comment (lukechurch) [inline src/lib/pipeline_configuration.py:78]: What is the logic for using the match value as opposed to the string value? The match values aren't in some sense 'canonical'? I could be persuaded either way...

**Review by lukechurch — COMMENTED** [review]
Review in progress

File: src/lib/pipeline_configuration.py (hunk 3 above)
Comment (as2388) [reply src/lib/pipeline_configuration.py:78]: string_value was for analysis output. 

We created match_values for this specific, separate purpose of needing to get from strings to codes.

File: src/lib/pipeline_configuration.py (hunk 2 above)
Comment (as2388) [reply src/lib/pipeline_configuration.py:48]: No, these are definitely Code objects.

Comment (as2388) [reply src/lib/pipeline_configuration.py:48]: so the code that has the ID associated with the label in the coda scheme

File: generate_outputs.py (hunk 1 above)
Comment (lukechurch) [reply generate_outputs.py]: ping

**Review by lukechurch — APPROVED** [review]
Thanks @as2388

File: generate_outputs.py (hunk 1 above)
Comment (as2388) [reply generate_outputs.py]: removed

**Pull request merged**